databases/gorm/categories.go

```diff
@@ -1,25 +1,61 @@
 package database
 
-import "github.com/arshamalh/blogo/models"
+import (
+	"errors"
+	"log"
+
+	"github.com/arshamalh/blogo/models"
+	"gorm.io/gorm"
+)
 
 func (gdb *gormdb) CheckCategoryExists(name string) bool {
-	results := gdb.db.Take(&models.Category{}, "name = ?", name)
-	return results.RowsAffected > 0
+	var category models.Category
+	err := gdb.db.Where("name = ?", name).First(&category).Error
+	exists := !errors.Is(err, gorm.ErrRecordNotFound)
+
+	if exists {
+		log.Printf("Category with name '%s' exists.", name)
+	} else {
+		log.Printf("Category with name '%s' does not exist.", name)
+	}
+
+	return exists
```

Comment:
we don't need logging here as it's not critical or important, we should also prevent extra and strict logging.

But just for you to know, it could be handled much simpler:
```go
if exists := results.RowsAffected > 0; exists {
   log.Gl.Info("category exist") // and extra fields for category id
   return exists // return right here so if exists, we don't go further
}
log.Gl.Info("category does not exist")
return false
```

Reply:
ok tanks

Reply:
ok thanks